content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,108 @@
+
+There are a bunch of variations of this pattern, the most popular ones include a Coordinator that would handle navigation and flow of the app relieving view controllers from this work.  
+
+For an example of this implementation you can check out [Rootstrap's iOS base](https://github.com/rootstrap/ios-base) a very helpful iOS project template to let you jump into your new project quickly. Check this [article](https://www.rootstrap.com/blog/2019/10/25/whats-great-about-our-ios-base/) to find out more about why we did that and how helpful it has been for us.  
+
+Whatever architecture pattern you decide on, it's very important to keep related functions together in sections organized with Marks. Some examples for controllers are: View Lifecycle, Layout, User Interaction, Protocol Implementation, etc. 
+
+It is also a nice practice to separate private functions into a private extension, so that a client at a glance at the class can differentiate the interface from the implementation.  
+
+## Resource Management 
+
+As we mention before we want to have a folder in the root of the project for the resources our app will need, there are many types of resources so its helpful to organize them in different folders according to their type, for example, you may have: property lists, images, gifs, JSONs... ant the list goes on.
+
+For images, Xcode provides a very helpful resource bundle named xassets. 
+Here we'll store the images our app uses, if your app has a fancy UI and uses loads of them is good to have them grouped in folders that describe their use. 
+
+Images may be heavy, and we want our apps to be as small in size as possible, so if we find yourself having to add the same image in different colors its recommended to use them as templates and color them in code instead of adding different versions of the same image.  
+
+Also, the naming of the resource itself should be descriptive enough so that you'll find it quickly and makes sense in the place you reference it. 
+
+## Storyboards and xibs vs Code
+
+Storyboards and xibs are great when you start learning to code in swift. 
+The visual help does a lot kickstarting your designs and helping you understand `NSLayoutConstraints`.
+Also if well handled, the way they represent the flow helps you understand a lot of the app by having a glance at it.
+
+However... They are heavy and often take a lot to load, connections between storyboard views and code are not checked in compiler time so you may have crashes if not careful, when working in a team the merge conflicts will be very hard to solve, its very annoying to review the autogenerated code in Pull Requests, it doesn't favor reuse and componentization, and most importantly there's not a "single source of truth" for your UI code. UI and navigation code will be divided between storyboards and your view controllers making it way harder to understand and maintain. 
```

Comment:
I would split this in multiple sentences or bullet points to avoid the really long sentence.